app/components/item-thumbnail-view.js

```diff
@@ -1,45 +1,101 @@
+import ImagePreview from "./image-preview";
+import { getWithDefault, computed } from "@ember/object";
 import { inject as service } from "@ember/service";
 import cloudinaryImage from "../mixins/cloudinary_image";
+import _ from "lodash";
 
-export default Component.extend(cloudinaryImage, {
-  loading: true,
+export default ImagePreview.extend(cloudinaryImage, {
   store: service(),
 
-  changedSrc: function() {
-    if (!this.is_cached(this.get("src"))) {
-      this.set("loading", true);
+  initializeLightgallery() {
+    var gallery = $(`#lightGallery${this.get("item.id")}`).data("lightGallery");
+    if (gallery) {
+      gallery.destroy();
     }
-  }.observes("src"),
 
-  didReceiveAttrs() {
-    this._super(...arguments);
-    this.set("cloudinaryId", this.get("cloudinaryId"));
-    this.set("src", this.generateUrl(500, 500, true));
-    const pkgTypeType = this.get("store").peekRecord(
-      "package_type",
-      this.get("item").attributes.package_type_id
+    var lightGalleryObj = $(`#lightGallery${this.get("item.id")}`).lightGallery(
+      {
+        mode: "lg-slide",
+        zoom: true,
+        download: false,
+        scale: 1,
+        hideControlOnEnd: true,
+        closable: true,
+        loop: true,
+        counter: true,
+        enableTouch: true,
+        enableDrag: true,
+        selector: ".imageZoom"
+      }
     );
-    this.set("pkgTypeType", pkgTypeType);
+    this.set("lightGalleryObj", lightGalleryObj);
   },
 
-  onLoad: function() {
-    this.set("loading", false);
+  safeGet(model, id) {
+    let record = this.get("store").peekRecord(model, id);
+    return getWithDefault(record, "name", "N/A");
   },
 
-  is_cached: function(src) {
-    var image = new Image();
-    image.src = src;
-    return image.complete;
-  },
+  packageType: computed("item", function() {
+    return this.safeGet(
+      "package_type",
+      this.get("item").attributes.package_type_id
+    );
+  }),
 
-  didInsertElement() {
-    var updateScreen = bind(this, this.onLoad);
-    this.$(".cl-item-image").on("load", updateScreen);
-  },
+  description: computed(
+    "item.id",
+    "i18n.locale",
+    "item.attributes.notes_zh_tw",
+    "item.attributes.notes",
+    function() {
+      const lang = this.get("i18n.locale");
+      const chineseDescription = (
+        this.get("item").attributes.notes_zh_tw || ""
+      ).trim();
+
+      if (lang === "zh-tw" && !!chineseDescription) {
+        return chineseDescription;
+      }
 
-  willDestroyElement() {
-    this.$(".cl-item-image").off("load");
-  }
+      return this.get("item").attributes.notes;
+    }
+  ),
+
+  condition: computed(
+    "item.id",
+    "item.attributes.donor_condition_id",
+    function() {
+      return this.safeGet(
+        "donor_condition",
+        this.get("item").attributes.donor_condition_id
+      );
+    }
+  ),
+
+  dimension: computed("item", "item.id", function() {
+    let length = _.get(this.get("item"), "attributes.length");
```

Comment:
for `length` you have used lodash get, and for `width` and `height` you have used the normal `get`
want to make that consistent?